### ISSUE TITLE
`data` can be briefly inconsistent with `tail` when the route path changes

### ISSUE THREAD

**Issue by 43081j** [PolymerElements/app-route#82]
Inside carbon-route (in `__tryToMatch`), `this.data` is set just before `this.tail`, as you see [here](https://github.com/PolymerElements/carbon-route/blob/master/carbon-route.html#L276).

This results in the inability to do routing logic on data changes because the current tail is incorrect (old).

For example, if you have a single route of `/:page` and use the data to control iron-pages, like so:

```html
<iron-pages selected="{{routeData.page}}" attr-for-selected="data-route">
    <my-page data-route="contact" route="{{route.tail}}"></my-page>
</iron-pages>
```

`my-page` will be selected as soon as `routeData.page` changes, but at that point, `route.tail` is the *old* tail. So, imagine in `my-page`, we do:

```javascript
observers: [
    '_onActive(active)'
],

_onActive: function(active) {
    if(active) {
        this.set('route.path', '/default');
    }
}
```

It will be attempting to set `/theOldPage/default` rather than `/contact/default`.

This can probably be worked around by observing `route.prefix` and checking if it equals `/contact` before setting the path. However, we should not know or care what path we are in, that is upto the parent to decide, so this workaround would be incorrect.

Would it be an issue to swap them around? Or do you know a better solution?

**Comment by rictic** [comment]
I think that my solution for #44 should address this. It involves hacking into the ugly details of the internals of the data binding system, but as the change is currently exists it will appear to the outside world as though `tail` and `data` change at the same time, atomically.

**Comment by TimvdLippe** [comment]
This issue has been fixed with #93 I suppose

**Comment by rictic** [comment]
Oh, good catch, yep!
